orbit/pkg/update/flag_runner.go

```diff
@@ -126,6 +126,160 @@ func (r *FlagRunner) DoFlagsUpdate() (bool, error) {
+func (r *ExtensionRunner) Execute() error {
+	log.Debug().Msg("starting extension runner")
+
+	ticker := time.NewTicker(r.opt.CheckInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-r.cancel:
+			return nil
+		case <-ticker.C:
+			log.Info().Msg("calling /config API to fetch/update extensions")
+			didExtensionsUpdate, err := r.DoExtensionConfigUpdate()
+			if err != nil {
+				log.Info().Err(err).Msg("ext update failed")
+			}
+			if didExtensionsUpdate {
+				log.Info().Msg("successfully updated/fetched extensions from /config API")
+				return nil
+			}
+		}
+		ticker.Reset(r.opt.CheckInterval)
+	}
+}
+
+// Interrupt is the oklog/run interrupt method that stops orbit when interrupt is received
+func (r *ExtensionRunner) Interrupt(err error) {
+	close(r.cancel)
+	log.Debug().Err(err).Msg(("interrupt extension runner"))
+}
+
+// DoExtensionConfigUpdate calls the /config API endpoint to grab extensions from Fleet
+// It parses the extensions, computes the local hash, and writes the binary path to extension.load file
+func (r *ExtensionRunner) DoExtensionConfigUpdate() (bool, error) {
+	// call "/config" API endpoint to grab orbit configs from Fleet
+	config, err := r.configFetcher.GetConfig()
+	if err != nil {
+		return false, fmt.Errorf("extensionsUpdate: error getting extensions config from fleet: %w", err)
+	}
+
+	extensionAutoLoadFile := filepath.Join(r.opt.RootDir, "extensions.load")
+	if len(config.Extensions) == 0 {
+		// Extensions from Fleet is empty
+		// this can be either because of:
+		// 1. the default state, where no extensions are configured to begin with, or
+		// 2. extensions were previously were configured, but now are deleted and reverted to empty state
+
+		// Handle case 1, where our autoload file does not exist, so there is nothing to update and no error
+		stat, err := os.Stat(extensionAutoLoadFile)
+		if errors.Is(err, os.ErrNotExist) {
+			log.Debug().Msg(extensionAutoLoadFile + " not found, nothing to update")
+			return false, nil
+		}
+
+		if stat.Size() > 0 {
+			// handle case 2: create/truncate the extensions.load file and let the runner interrupt, so that
+			// osquery can't startup without the extensions that were previously loaded
+			_, err := os.Create(extensionAutoLoadFile)
+			if err != nil {
+				return false, fmt.Errorf("extensionsUpdate: error creating file %s, %w", extensionAutoLoadFile, err)
+			}
+			// we want to return true here, and restart with the empty extensions.load file
+			// so that we "unload" the previously loaded extensions
+			return true, nil
+		}
+		return false, nil
+	}
+
+	type ExtensionInfo struct {
+		Platform string `json:"platform"`
+		Channel  string `json:"channel"`
+		FileName string `json:"file_name"`
+	}
+
+	var data map[string]ExtensionInfo
+	err = json.Unmarshal(config.Extensions, &data)
+	if err != nil {
+		return false, fmt.Errorf("error unmarshing json extensions config from fleet: %w", err)
+	}
+
+	var sb strings.Builder
+	for k, v := range data {
```

Comment:
Something slightly more descriptive could be nice. Maybe this?
```suggestion
	for ext, opt := range data {
```

Reply:
Yes, agree, updated to use slightly more descriptive names